File: java-suite/src/main/java/de/fraunhofer/iem/classesAndObjects/classes/constructors/multipleConstructors/Employee.java
package de.fraunhofer.iem.classesAndObjects.classes.constructors.multipleConstructors;

/**
 * Demonstrates the multiple constructor in Java
 *
 * @author Ranjith Krishnamurthy
 */
public class Employee {
    String firstName;
    String lastName;
    int age;

    Employee(String firstName) {
        System.out.println("Initializer block");
        this.firstName = firstName;
        System.out.println("Completed Initializer block");
    }

    Employee(String firstName, String lastName) {
        this(firstName);
        this.lastName = lastName;
    }

    Employee(String firstName, String lastName, int age) {
        this(firstName, lastName);
        this.age = age;
    }
}

File: kotlin-suite/src/main/kotlin/de/fraunhofer/iem/classesAndObjects/classes/constructors/secondaryConstructor/justSecondaryConstructor/Employee.kt
package de.fraunhofer.iem.classesAndObjects.classes.constructors.secondaryConstructor.justSecondaryConstructor

/**
 * Demonstrates the secondary constructor without the primary constructor
 *
 * @author Ranjith Krishnamurthy
 */
class Employee {
    var firstName: String = ""
    var lastName: String = ""
    var age: Int = -100

    init {
        println("Initializer block")
        println("Completed Initializer block")
    }

    constructor(firstName: String, lastName: String, age: Int) {
        println("Secondary Constructor")
        this.firstName = firstName
        this.lastName = lastName
        this.age = age
        println("Completed Secondary Constructor")
    }

    // Just to demonstrate the order of the init block in the primary constructor in the Java bytecode
    init {
        println("Second Initializer block")
        println("Completed Second Initializer block")
    }
}